lib/internal/main/check_syntax.js

```diff
@@ -36,18 +37,29 @@ if (process.argv[1] && process.argv[1] !== '-') {
 
   markBootstrapComplete();
 
-  checkSyntax(source, filename);
+  loadESMIfNeeded(() => checkSyntax(source, filename));
 } else {
   markBootstrapComplete();
 
-  readStdin((code) => {
+  loadESMIfNeeded(() => readStdin((code) => {
     checkSyntax(code, '[stdin]');
-  });
+  }));
 }
 
-async function checkSyntax(source, filename) {
+function loadESMIfNeeded(cb) {
   const { getOptionValue } = require('internal/options');
-  let isModule = false;
+  const shouldLoadESM = getOptionValue('--import').length > 0;
```

Comment:
nit: this name also seems a lot more encompassing than it actually is.

```suggestion
  const hasModulePreimport = getOptionValue('--import').length > 0;
```